Motoko team suggestions (#5062)
dfinity/motoko: doc/md/fundamentals/3-declarations/2-function-declarations.md

        return 1;
    };
// Calls itself with factorial(n - 1), reducing n until reaching 0.
    n * factorial(n - 1);
}
```

## Generic functions

> Motoko supports generic functions, which allow you to write reusable logic that works with any type.

Reviewer: ```suggestion
Generic functions are used to write reusable logic that works with any type.
```
